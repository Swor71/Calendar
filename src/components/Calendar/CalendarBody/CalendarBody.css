.calendar-body-wrapper {
  display: grid;
  grid-template-columns: repeat(7, minmax(50px, 1fr));
  grid-template-rows: repeat(auto-fill, 1fr);
  grid-gap: 1px;
}

[class*='day'] {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 50px;
  min-width: 50px;
}

.weekday {
  color: #000;
  font-size: .9em;
  font-weight: bold;
}

.day {
  border: 1px solid #dee0e4;
  color: #dee0e4;
}
