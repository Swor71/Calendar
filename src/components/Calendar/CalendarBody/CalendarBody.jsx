import React, { PureComponent } from 'react';

class CalendarBody extends PureComponent {
  render() {
    return (
      <div>

      </div>
    );
  }
}

export default CalendarBody;
