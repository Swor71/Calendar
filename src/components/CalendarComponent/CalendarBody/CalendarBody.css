.calendar-body__wrapper {
  display: grid;
  grid-template-columns: repeat(7, minmax(50px, 1fr));
  grid-template-rows: repeat(auto-fill, 1fr);
  margin-top: 10px;
}

[class*='day'] {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 50px;
  min-width: 50px;
}

.weekday {
  font-size: .9em;
  font-weight: bold;
}

.emptyday {
  background-color: transparent;
}

.day {
  border: 1px solid #dee0e4;
  color: #dee0e4;
  cursor: pointer;
  transition: .1s;
}

.day:hover {
  color: #fff;
  background-color: paleturquoise;
}
