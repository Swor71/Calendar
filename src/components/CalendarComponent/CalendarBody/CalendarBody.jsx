import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { dayList, dayOfMonth, currentMonth } from '../../../utils/dateFormats';

import './CalendarBody.css';

class CalendarBody extends Component {
  static propTypes = {
    calendarData: PropTypes.object,
    availableDates: PropTypes.arrayOf(PropTypes.object),
    onHandleSelectDate: PropTypes.func,
  };

  static defaultProps = {
    calendarData: {
      daysInMonth: [],
      month: '',
      emptyBlocks: [],
    },
    availableDates: [],

  };

  render() {
    const { calendarData: { month, daysInMonth, emptyBlocks }, availableDates, onHandleSelectDate } = this.props;

    let daysAndAvailability = [];
    daysInMonth.forEach((item, i) => {
      daysAndAvailability.push(Object.assign({}, item, availableDates[i]));
    });

    return (
      <div className='calendar-body__wrapper'>
        {dayList.map(day => <div className='weekday' key={day} >{day}</div>)}
        {emptyBlocks.map(block => <div className="emptyday" key={block}></div>)}
        {daysAndAvailability.map(day => {
          const { available, dayNumber } = day;
          const disabledClass = (month >= currentMonth) && available ? '' : 'disabled';
          const today = (dayOfMonth === dayNumber) && (currentMonth === month) ? 'today' : '';
          return (
            <button
              key={dayNumber}
              className={`day ${disabledClass} ${today}`}
              disabled={month < currentMonth || !available}
              onClick={() => onHandleSelectDate(dayNumber)}
            >{dayNumber}</button>
          );
        })}
      </div>
    );
  }
}
export default CalendarBody;
