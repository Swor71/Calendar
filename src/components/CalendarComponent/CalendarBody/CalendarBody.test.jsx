import React from 'react';
import CalendarBody from './CalendarBody';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('CalendarBody', () => {
  it('renders the component', () => {
    const component = shallow(<CalendarBody />);
    expect(component.length).toEqual(1);
  });

  it('renders correctly', () => {
    const component = renderer
      .create(<CalendarBody />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
