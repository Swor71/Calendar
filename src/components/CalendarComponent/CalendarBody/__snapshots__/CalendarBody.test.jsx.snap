// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarBody renders correctly 1`] = `
<div
  className="calendar-body__wrapper"
>
  <div
    className="weekday"
  >
    Su
  </div>
  <div
    className="weekday"
  >
    Mo
  </div>
  <div
    className="weekday"
  >
    Tu
  </div>
  <div
    className="weekday"
  >
    We
  </div>
  <div
    className="weekday"
  >
    Th
  </div>
  <div
    className="weekday"
  >
    Fr
  </div>
  <div
    className="weekday"
  >
    Sa
  </div>
</div>
`;
